Trailing empty lines in the batches text file makes the pipeline fail

If the batches files given as input has trailing empty lines, the pipeline will fail. For example, if I add an empty line to `data/batches_small.txt`:
```
actinobacillus_pleuropneumoniae__01
aeromonas_salmonicida__01
bacillus_anthracis__01

```
And run `make test`, `mof-search` pipeline will fail because it interprets the empty line as a batch `''`:

```
$ make test
snakemake --jobs all --rerun-incomplete --printshellcmds --keep-going --use-conda --resources max_download_threads=8 max_io_heavy_threads=8 max_ram_mb=12288 -j 99999 --config batches=data/batches_small.txt -- download  # download is not benchmarked
Config file config.yaml is extended by additional config specified via the command line.
Batches: ['', 'actinobacillus_pleuropneumoniae__01', 'aeromonas_salmonicida__01', 'bacillus_anthracis__01']
Query files: ['queries/query.fa', 'queries/reads_4.fa', 'queries/reads_3.fasta', 'queries/reads_2.fq', 'queries/reads_1.fastq']
Building DAG of jobs...
MissingInputException in line 217 of /home/leandro/git/mof-search/Snakefile:
Missing input files for rule download:
    affected files:
        asms/.tar.xz
        cobs/.cobs_classic.xz
make: *** [Makefile:22: test] Error 1
```

yes:
```
$ make test
snakemake download --cores all --rerun-incomplete --printshellcmds --keep-going --use-conda --resources max_download_threads=8 max_io_heavy_threads=8 max_ram_mb=12288 -j 99999 --config batches=data/batches_small.txt  # download is not benchmarked
Config file config.yaml is extended by additional config specified via the command line.
Batches: ['', '', 'actinobacillus_pleuropneumoniae__01', 'aeromonas_salmonicida__01', 'bacillus_anthracis__01']
Query files: ['input/reads_4.fa', 'input/reads_3.fasta', 'input/reads_2.fq', 'input/reads_1.fastq']
Building DAG of jobs...
MissingInputException in rule download in file /home/leandro/git/mof-search/Snakefile, line 224:
Missing input files for rule download:
    affected files:
        cobs/.cobs_classic.xz
        asms/.tar.xz
make: *** [Makefile:36: test] Error 1
```

--- a/Snakefile
+++ b/Snakefile
@@ -31,7 +31,7 @@ def get_all_query_filenames():
 
 def get_batches():
     with open(config["batches"]) as fin:
-        return sorted([x.strip() for x in fin])
+        return list(sorted(filter(len, map(str.strip, fin))))
 
 
 def get_filename_for_all_queries():

